src/api/controllers/BlockchainController.ts

```diff
@@ -0,0 +1,83 @@
+import {FastifyRequest, FastifyReply} from 'fastify';
+import PointSDKController from './PointSDKController';
+import blockchain from '../../network/blockchain';
+import permissionStore from '../../permissions/PermissionStore';
+
+export type RPCRequestBody = {
+    method: string;
+    params?: unknown[];
+};
+
+type PermissionHandlerFunc = (
+    req: FastifyRequest
+) => Promise<{
+    status: number;
+    result: unknown;
+}>;
+
+const permissionHandlers: Record<string, PermissionHandlerFunc> = {
+    wallet_requestPermissions: async (req: FastifyRequest) => {
+        try {
+            const dappDomain = req.headers.origin;
+            if (!dappDomain) {
+                return {status: 400, result: {message: '`Origin` header is required.'}};
+            }
+
+            const {params} = req.body as RPCRequestBody;
+            const address = blockchain.getOwner();
+            const allowedMethods = params ? Object.keys(params[0] as Record<string, object>) : [];
+
+            const id = await permissionStore.upsert(dappDomain, address, allowedMethods);
+            return {status: 200, result: {permissionsId: id}};
+        } catch (err) {
+            return {status: 400, result: err};
+        }
+    },
+    wallet_getPermissions: async (req: FastifyRequest) => {
+        const dappDomain = req.headers.origin;
+        if (!dappDomain) {
+            return {status: 400, result: {message: '`Origin` header is required.'}};
+        }
+
+        const address = blockchain.getOwner();
+        const permissions = await permissionStore.get(dappDomain, address);
+        return {status: 200, result: permissions || null};
```

Comment:
the `result` is `permissions || null` because `permissions` could be `undefined`, and Fastify removes the key when the value is `undefined`. I thought it would cleaner for the client to always get a `permissions` key in the response, hence the `null`.